lib/message.js

```diff
@@ -0,0 +1,44 @@
+'use babel';
+/* @flow */
+
+type Range = Array<Array<number>>
+type LinterTrace = { type: 'Trace', text?: string, html?: string, filePath?: string, range?: Range, class?: string }
+type LinterMessage = { type: string, text?: string, html?: string, filePath: string, range: Range, trace?: Array<LinterTrace> }
+type FlowMessage = { descr: string, level: string, path: string, start: number, end: number, line: number, endline: number }
+
+function extractRange(message: FlowMessage): Range {
+  return [
+    [message.line - 1, message.start - 1],
+    [message.endline - 1, message.end],
+  ];
+}
+
+function flowMessageToTrace(message: FlowMessage): LinterTrace {
+  return {
+    type: 'Trace',
+    text: message.descr,
+    filePath: message.path,
+    range: extractRange(message),
+  };
+}
+
+function flowMessageToLinterMessage({ message: messages }): LinterMessage {
+  const first = messages[0];
+
+  return {
+    type: first.level,
+    text: messages.map(msg => msg.descr).join('. '),
```

Comment:
@steelbrain nuclide do the same to render message in Panes, [see here](https://github.com/facebook/nuclide/blob/746090718fe19222a88c1d34e23c06428dc294c9/pkg%2Fnuclide%2Fdiagnostics%2Fui%2Flib%2FDiagnosticsPane.js#L75)